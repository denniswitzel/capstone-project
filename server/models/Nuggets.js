import mongoose from 'mongoose'

const NuggetsSchema = mongoose.Schema({
    category: String,
    title: String,
    ingredients: String,
    allergies: Array,
    image: String,
    rating: Array
})

export default mongoose.model('nuggets', NuggetsSchema, 'nuggets')