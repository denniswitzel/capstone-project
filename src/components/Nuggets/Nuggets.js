import React from 'react'
import nuggets from '../../mocks/nuggets.json'
import Headline from '../Headline/Headline'
import Product from '../Products/Product'
import PropTypes from 'prop-types'

Nuggets.propTypes = { 
    nuggets: PropTypes.arrayOf(PropTypes.object).isRequired,
    title: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
}

export default function Nuggets(){
    return(
        <>
        <Headline headline="Nuggets" />
        {nuggets.map((nugget, index) => (
        <Product key={index} title={nugget.title} image={nugget.image} />
        ))}
        </>
    )
}
