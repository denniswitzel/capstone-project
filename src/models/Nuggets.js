const mongoose = require('mongoose')

const NuggetsSchema = mongoose.Schema({
    category: String,
    title: String,
    ingredients: String,
    allergies: Array,
    image: String,
    rating: Array
})

module.exports = mongoose.model('nuggets', NuggetsSchema, 'nuggets')