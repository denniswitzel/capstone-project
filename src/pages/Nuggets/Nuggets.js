import React from 'react'
import nuggets from '../../mocks/nuggets.json'
import Product from '../../components/Products/Product'
import Headline from '../../components/Headline/Headline'
import PropTypes from 'prop-types'

Nuggets.propTypes = { 
    nuggets: PropTypes.arrayOf(PropTypes.object),
    title: PropTypes.string,
    image: PropTypes.string,
}

export default function Nuggets(){
    return(
        <>
        <Headline headline="Nuggets" />
        {nuggets.map((nugget, index) => (
        <Product key={index} title={nugget.title} image={nugget.image} />
        ))}
        </>
    )
}
