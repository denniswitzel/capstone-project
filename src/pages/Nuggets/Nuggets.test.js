import React from 'react'
import renderer from 'react-test-renderer'
import Nuggets from '../Nuggets/Nuggets'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

describe('Nuggets', () => {
    const title = 'Like meat - Like Nuggets'
    const image = 'https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/nuggets/Like-meat-Nuggets.png'

    it('display title and image', () => {
        const { getByText } = render(<Nuggets title={title} image={image}/>)
        expect(getByText(title)).toBeInTheDocument()
        expect(document.querySelector("img").getAttribute("src")).toBe(image)
    })

    it('renders correctly', () => {
        const tree = renderer.create(<Nuggets />)
        expect(tree).toMatchSnapshot()
      })
  })