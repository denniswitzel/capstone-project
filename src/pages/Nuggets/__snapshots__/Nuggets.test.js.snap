// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nuggets renders correctly 1`] = `
Array [
  <section
    className="sc-AxhCb djfBDV"
  >
    <svg
      className="sc-AxgMl kRQlAl"
      onClick={[Function]}
    >
      back-arrow.svg
    </svg>
    <h2
      className="sc-AxhUy gghUbh"
    >
      Nuggets
    </h2>
  </section>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Like meat - Like Nuggets
    </h3>
    <img
      alt="Like meat - Like Nuggets"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/nuggets/Like-meat-Nuggets.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Rügenwalder - Mühlen Nuggets
    </h3>
    <img
      alt="Rügenwalder - Mühlen Nuggets"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/nuggets/Ruegenwalder-Muehlen-Nuggets.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Vegetarian Butcher - Beflügel Nuggets
    </h3>
    <img
      alt="Vegetarian Butcher - Beflügel Nuggets"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/nuggets/The-Vegetarian-Butcher-Vegane-Befl%C3%BCggel-Nuggets.png"
    />
  </div>,
  <div
    className="sc-AxjAm jTSiGe"
  >
    <h3
      className="sc-AxirZ FzsEO"
    >
      Vivera - Vegane Nuggets
    </h3>
    <img
      alt="Vivera - Vegane Nuggets"
      className="sc-AxiKw erJOzJ"
      src="https://raw.githubusercontent.com/denniswitzel/capstone-project/master/src/images/nuggets/Vivera-Vegane-Nuggets.png"
    />
  </div>,
]
`;
